power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -967,6 +978,9 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
 
         regulator.transformer.apply([&](transformer_c auto const& transformer) { // NOSONAR
             using TransformerType = std::remove_cvref_t<decltype(transformer)>;
+            if (!is_regulated_transformer_connected<TransformerType>(regulator, state)) {
```

Comment:
Maybe we shouldremove the unconnected regulated transformer from the ranking itself. Hence, this connected by topology check instead of the status checks. This would alter the ranking of transformers but for the better.

Reply:
More digging gives the surprising reality check that the `topo_comp_coup` member of `state` remains `empty` during trafo ranking and is only updated after/during iterations. This renders the impossibility to retrieve necessary information in the transformer ranking stage. Therefore, unless we do more fundamental changes, the current way might be the only reasonable solution given the present state of code.

Reply:
the entire topology step is done only during the first calculation. we can change this, of course, with significant changes (my guess is that these are what you mean)

however, it should be possible to determine which nodes/regulators are not energized because they obtain infinite path length to the nearest source during transformer ranking (with the only overhead compared to skipping them entirely being the creation)

Reply:
while something we need to keep in our minds, it is out of scope of this PR